src/providers/README.md

```diff
@@ -2,3 +2,85 @@
 
 This directory contains providers contributing additional bindings, for example
 custom sequence actions.
+
+## Overview
+
+A provider is a class that provides a `value()` function. This function is called `Context` when another entity requests a value to be injected.
```

Comment:
+1 for this short introduction. Please add a link to the doc page which explains the concept of providers (and maybe sequence actions too?) in more details.

Reply:
+1 to have links.